### PR TITLE
Add Fast DDS RTPS-layer API reference [8714]

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/521/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/524/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/526/)

File: docs/fastdds/api_reference/rtps/common/Types/types_toc.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
+.. _api_rtps_common_types:
+
+Types
+--------------------
+
+.. toctree::
+
+    /fastdds/api_reference/rtps/common/Types/BuiltinEndpointSet_t
```
Comment (EduPonz) [inline docs/fastdds/api_reference/rtps/common/Types/types_toc.rst:8]: I believe `inline std::ostream& operator<<(std::ostream& output, const ProtocolVersion_t& pv)` is missing

File: docs/spelling_wordlist.txt [hunk 2]
```diff
@@ -132,7 +134,13 @@ runtime
 sanitizers
 scalability
 scalable
+<<<<<<< HEAD
 simpleEDP
+=======
+seqNum
+SequenceNumber
+sequencenumbers
+>>>>>>> 80221d21... Refs #8612: Added the remaining common settings.
```
Comment (EduPonz) [inline docs/spelling_wordlist.txt:143]: Ups!

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/536/)

**Pull request merged**